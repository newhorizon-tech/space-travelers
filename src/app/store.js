import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import rocketsReducer from '../features/rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    counter: counterReducer,
    rockets: rocketsReducer,
  },
});

export default store;
