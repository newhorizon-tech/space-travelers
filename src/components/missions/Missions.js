import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { missionsFetch } from '../../redux/ missions/missionsReducer';
import Mission from './Mission';

function Missions() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions_reducer);

  useEffect(() => { dispatch(missionsFetch()); }, []);

  return (
    <div className="mission-container">
      <div className="missionRaw">
        <div className="missionName">Mission</div>
        <div className="missionDescription">Description</div>
        <div className="missionStatus missionStatus-border">Status</div>
        <div className="missionStatus">Status</div>
      </div>
      {
        missions.map((mission) => <Mission singleMission={mission} key={mission.mission_id} />)
      }
    </div>
  );
}

export default Missions;
