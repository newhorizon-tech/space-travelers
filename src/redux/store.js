import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import missionsReducer from './ missions/missionsReducer';

const reducer = combineReducers({

  missions_reducer: missionsReducer,

});

const store = createStore(
  reducer,
  applyMiddleware(thunk),
);

export default store;
