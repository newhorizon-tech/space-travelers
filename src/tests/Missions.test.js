import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Missions from '../components/missions/Missions';

describe('Mission Component test', () => {
  it('Renders Mission Row', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Missions />

      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
