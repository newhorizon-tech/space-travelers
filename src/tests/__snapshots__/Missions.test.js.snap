// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Component test Renders Mission Row 1`] = `
<div
  className="mission-container"
>
  <div
    className="missionRaw"
  >
    <div
      className="missionName"
    >
      Mission
    </div>
    <div
      className="missionDescription"
    >
      Description
    </div>
    <div
      className="missionStatus missionStatus-border"
    >
      Status
    </div>
    <div
      className="missionStatus"
    >
      Status
    </div>
  </div>
</div>
`;
